src/snowflake/cli/_plugins/workspace/manager.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+from typing import Dict
+
+from snowflake.cli.api.entities.common import EntityActions
+from snowflake.cli.api.exceptions import InvalidProjectDefinitionVersionError
+from snowflake.cli.api.project.schemas.entities.entities import (
+    Entity,
+    v2_entity_model_to_entity_map,
+)
+from snowflake.cli.api.project.schemas.project_definition import DefinitionV20
+
+
+class WorkspaceManager:
+    """
+    Instantiates entity instances from entity models, providing higher-order functionality on entity compositions.
+    """
+
+    def __init__(self, project_definition: DefinitionV20, project_root: Path):
+        if not project_definition.meets_version_requirement("2"):
+            raise InvalidProjectDefinitionVersionError(
+                "2.x", project_definition.definition_version
+            )
+        self._entities_cache: Dict[str, Entity] = {}
+        self._project_definition = project_definition
+
+    def get_entity(self, entity_id: str):
+        if entity_id not in self._entities_cache:
+            if entity_id not in self._project_definition.entities:
+                raise ValueError(f"No such entity ID: {entity_id}")
+            entity_model_cls = self._project_definition.entities[entity_id].__class__
+            entity_cls = v2_entity_model_to_entity_map[entity_model_cls]
```

Comment:
Should we consider implementing something like this on `EntityModelBase`?
```py
model.get_entity_class()
```

In this way we don't have to implement a mapping and we keep the relation defined in one place.
```
class ApplicationEntityModel(EntityModelBase)
   __model_class = ApplicationPackageEntity
```

Later we can change `get_entity_class` to `get_entity` that would return instance of entity build from the model? WDYT?

Reply:
+1

Reply:
Yes, that's actually what I started with, but circular deps got in the way and it wasn't trivial to untangle them. Let's do it as a separate PR together with `get_entity()`?